### PR TITLE
bpo-40369: Use PEP 590 vectorcall to speed up GenericAlias

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue40369

### CODE REVIEW

File: Objects/genericaliasobject.c [hunk 1]
```diff
@@ -438,19 +438,33 @@ static PyGetSetDef ga_properties[] = {
...
 ga_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
-    if (kwds != NULL && PyDict_GET_SIZE(kwds) != 0) {
-        PyErr_SetString(PyExc_TypeError, "GenericAlias does not support keyword arguments");
+    if (!_PyArg_NoKwnames("GenericAlias", kwds)) {
         return NULL;
     }
-    if (PyTuple_GET_SIZE(args) != 2) {
-        PyErr_SetString(PyExc_TypeError, "GenericAlias expects 2 positional arguments");
+    if (!_PyArg_CheckPositional("GenericAlias", PyTuple_GET_SIZE(args), 2, 2)) {
```
Comment (gvanrossum) [inline Objects/genericaliasobject.c:444]: I like these specialized error checks, but I doubt the vector call helps much. I'd rather try adding a cache like we have in typing.py.

Comment (vstinner) [reply Objects/genericaliasobject.c:444]: I like the idea of a cache. In term of performance, it's usually hard to beat a cache ;-)

Comment (corona10) [reply Objects/genericaliasobject.c:444]: @gvanrossum @vstinner cc @serhiy-storchaka 
I agree with guido's opinion, hmm looks like this PR should be closed.
Is it okay to open the sperate PR for specialized error checks?

Comment (vstinner) [reply Objects/genericaliasobject.c:444]: You can open a separated PR changes on the error code.

Yeah, I suggest to close this PR.

Comment (corona10) [reply Objects/genericaliasobject.c:444]: > You can open a separated PR changes on the error code.

Thanks, I will do.

**Pull request closed without merging**